text/trace/0000-error_flagging.md

```diff
@@ -0,0 +1,86 @@
+A second exception is supporting analysis tools which require explicit error flagging in the data which they receive. In this case, an operator must be able to apply an error flagging schema at some point during the OTLP data processing pipeline.
+ 
+## Explanation
+The following changes add several missing features required for proper error reporting, and are completely backwards compatible with OpenTelemetry today.
+ 
+### Status Codes
+The following status codes are added to our current  schema.
+ 
+* `DEFAULT` No status has been set. Any errors must be detected by the analysis tool. (This replaces `OK`.)
+* `ERROR` Instrumentation has marked a span as an error. (This replaces `UNKNOWN`.)
+* `OK_OVERRIDE` The user has provided an override. The span should NOT be flagged as an error, regardless of other analysis.
+* `ERROR_OVERRIDE` The user has provided an override. The span SHOULD be flagged as an error, regardless of other analysis.
+ 
+(Note that our current status codes include a long list of error types. We may choose to keep  them, change them, or drop them in favor of a single `ERROR` code. How many error types we have is not relevant to this proposal.)
+ 
+`OK_OVERRIDE` and `ERROR_OVERRIDE` are special status codes. These are explicit overrides provided by the end user, and should never be set by shared instrumentation. They should only be set by the application developer (via application code), or by the operator (via the collector).
+ 
+Analysis tools are free to disregard status codes, in favor of their own approach to error analysis. However it is strongly suggested that analysis tools handle `OK_OVERRIDE` and `ERROR_OVERRIDE`, as these are explicitly set by the end user and contain valuable information.
+ 
+ 
+### Error Mapping Schema
+As part of the specification, OpenTelemetry provides a canonical mapping of semantic conventions to status codes. This removes any ambiguity as to what OpenTelemetry ships with out of the box.
+ 
+### Error Processor
+The collector will provide a processor and a configuration language to make adjustments to this error mapping schema. This provides the flexibility and customization needed for real world scenarios.
+ 
+### Convenience methods
+As a convenience, OpenTelemetry provides helper functions for adding semantic conventions and exceptions to a span. These helper functions will also set the correct status code. This simplifies the life of the instrumentation author, and helps ensure compliance and data quality.
+
+Note that these convenience methods simply wire together multiple API calls. They should live in a helper package, and should not be directly added to existing API interfaces. Given how many semantic conventions we have, there will be a pile of them.
+ 
+ 
+## Internal details
+Except for the renaming of two status codes, this proposal is backwards compatible with existing code, protocols, and the OpenTracing bridge.
```

Comment:
>this proposal is backwards compatible with existing code, protocols

For this to be true we need to keep all existing status codes and assign OK_OVERRIDE and ERROR_OVERRIDE numeric values that were previously unused (by numeric values I mean the values we use record them on the wire).

If we decide to remove existing (old) status codes as it is suggested as an option in the proposal (and which I think we should do) then it is not completely clear how we ensure the backward compatibility. Do we mark the previously used status codes and their corresponding numeric values on the wire as obsolete and disallow their usage in the generation side? Should the Collector continue to treat the obsolete codes seen on its input as errors?

A possible approach here which I think is reasonable is that we don't try to keep backward compatibility. We just break it and remove the old status codes from the API and don't care about the numeric values on the wire. I am not aware of existing production uses that may be affected by this and even if they are then perhaps it is one the things we break before the GA.

Reply:
Since we're now moving this to a separate `user_override` flag, we will no longer need to change the status codes as part of this OTEP. I do agree that it will be tricky to remove some of the status codes. I'm not sure how bad the breakage would be.